ddp-workspace/projects/ddp-sdk/src/lib/services/activity/activitySuggestionBuilder.service.ts

```diff
@@ -81,22 +87,30 @@ export class ActivitySuggestionBuilder {
 
     private getIncludedSuggestions(suggestions: Array<string>): TextSuggestionProvider {
         return (value$: Observable<string>) => value$.pipe(
-            map(value => this.findIncludedSuggestions(value, suggestions))
+            map(value => this.buildTextSuggestions(value, suggestions))
         );
     }
 
-    private findIncludedSuggestions(value: string, suggestions: Array<string>): Array<TextSuggestion> {
-        const lowerCaseValue = value.toLowerCase();
+    private getInstitutionSuggestions(): TextSuggestionProvider {
+        return (value$: Observable<string>) => value$.pipe(
+            switchMap(value => this.institutionService.getSummary(of(value), SUGGESTION_LIMIT).pipe(
+                map(institutions => institutions.map(institution => institution.name)),
+                map(names => this.buildTextSuggestions(value, names))
+            ))
+        );
+    }
+
+    private buildTextSuggestions(value: string, suggestions: Array<string>): Array<TextSuggestion> {
```

Comment:
This method turned out to be just what we needed for the institution autocompletions, so I refactored it and renamed it to be a bit more general.